linkerd/app/inbound/src/policy/store.rs

```diff
@@ -82,26 +74,35 @@ impl<S> Store<S> {
         // `idle_timeout` to prevent holding policy watches indefinitely for
         // ports that are generally unused.
         let cache = {
-            let opaque =
-                opaque_ports
-                    .into_iter()
-                    .flat_map(|config::OpaquePorts { ports, policy }| {
-                        let policy = Self::spawn_default(DefaultPolicy::Allow(policy));
-                        ports.into_iter().map(move |port| (port, policy.clone()))
-                    });
             let rxs = ports.into_iter().map(|port| {
                 let discover = discover.clone();
                 let default = default.clone();
                 let rx = info_span!("watch", port)
                     .in_scope(|| discover.spawn_with_init(port, default.into()));
                 (port, rx)
             });
-            Cache::with_permanent_from_iter(idle_timeout, rxs.chain(opaque))
+            Cache::with_permanent_from_iter(idle_timeout, rxs)
         };
+        Self::new(cache, default, Some(discover), opaque_ports)
+    }
+
+    fn new(
```

Comment:
Prefer to make it clear it spawns things.

```suggestion
    fn spawn(
```